support/sier-codec/src/json.rs

```diff
@@ -0,0 +1,69 @@
+use crate::{Error, Object, StructDef, Type, Value};
+use serde_json::Value as SerdeValue;
+use std::sync::Arc;
+
+pub fn transform_serde_value<'a>(
+    value: &SerdeValue,
+    type_: &'a Type,
+) -> Result<Value<'a>, Error<'a>> {
+    match value {
+        SerdeValue::Null => unimplemented!(),
+        SerdeValue::Bool(_bool) => unimplemented!(),
+        SerdeValue::Number(number) => {
+            if type_ == &Type::U64 {
```

Comment:
I am a big fan of early returns. For this case, I'd change the code to something like:

```rs
if *type_ != Type::U64 {
  return Err(...);
}

let n = number.as_u64().ok_or_else(|| unimplemented!())?;
Ok(Value::U64(n))
```

In general, it makes it very easy to tell how you're handling incorrect cases. It also keeps the code flat and more readable.

Reply:
How should this be done for the SerdeValue::Array and SerdeValue::Struct? For the String and Number I agree.